src/GameLoop.tsx

```diff
@@ -55,223 +59,214 @@ const stateMachine = createMachine({
 let lockStart = null;
-
-// The amount of time it takes before a block locks in place.
-const lockMax = 1500;
+const lockMax =
+    1500; /* The amount of time it takes before a block locks in place. */
 let matchAnimStart = null;
 const matchAnimLength = 750;
 let isMatchChaining = false;
 let isPlayerMovementEnabled = false;
-
 let didInstantDrop = false;
 
-function updatePlayerPos(
-    playerPhysics: PlayerPhysics,
-    boardPhysics: BoardPhysics,
-    { keyCode, repeat }: { keyCode: number; repeat: boolean },
-): void {
-    if (!isPlayerMovementEnabled) {
-        return;
-    }
-    const board = boardPhysics.boardCellMatrix;
-    const r = playerPhysics.pos[0];
-    const c = playerPhysics.pos[1];
-    const areTargetSpacesEmpty = (dr, dc) =>
-        playerPhysics.adjustedCells.every((cell) => {
-            return board[cell.r + dr][cell.c + dc].char === EMPTY;
-        });
-    if (keyCode === 37) {
-        // Left
-        if (
-            playerPhysics.isInCBounds(
-                playerPhysics.getAdjustedLeftmostC() - 1,
-            ) &&
-            // Ensure blocks don't cross over to ground higher than it, regarding interpolation.
-            (!ENABLE_SMOOTH_FALL ||
-                playerPhysics.isInRBounds(
-                    playerPhysics.getAdjustedBottomR() +
-                        Math.ceil(interp.val / interpMax),
-                )) &&
-            areTargetSpacesEmpty(
-                Math.ceil(ENABLE_SMOOTH_FALL ? interp.val / interpMax : 0),
-                -1,
-            )
-        ) {
-            playerPhysics.setPos(r, c - 1);
-            playerPhysics.hasMoved = true;
-        }
-    } else if (keyCode === 39) {
-        // Right
-        if (
-            playerPhysics.isInCBounds(
-                playerPhysics.getAdjustedRightmostC() + 1,
-            ) &&
-            // Ensure blocks don't cross over to ground higher than it, regarding interpolation.
-            (!ENABLE_SMOOTH_FALL ||
-                playerPhysics.isInRBounds(
-                    playerPhysics.getAdjustedBottomR() +
-                        Math.ceil(interp.val / interpMax),
-                )) &&
-            areTargetSpacesEmpty(
-                Math.ceil(ENABLE_SMOOTH_FALL ? interp.val / interpMax : 0),
-                1,
-            )
-        ) {
-            playerPhysics.setPos(r, c + 1);
-            playerPhysics.hasMoved = true;
-        }
-    } else if (keyCode === 40) {
-        // Down
-        if (repeat) {
-            // TODO: Handle repeated downkey.
-        }
-        if (
-            playerPhysics.getAdjustedBottomR() + 1 < BOARD_ROWS &&
-            areTargetSpacesEmpty(1, 0)
-        ) {
-            if (ENABLE_SMOOTH_FALL) {
-                interp.val += interpRate * interpKeydownMult;
-            } else {
-                playerPhysics.setPos(r + 1, c);
-            }
+export function GameLoop() {
+    const [boardPhysics, _setBoardPhysics] = useState(
```

Comment:
If you're not using the state or setState, something is likely wrong.

Reply:
I'll fix this later- put into #29

Reply:
Okay